### PR TITLE
GOVUKAPP 103 - Added snapshot tests

### PR DESCRIPTION
**Pull request by batesyuk**
Adding the ability to snapshot test views.
I added some initial tests, but the views aren't fully designed yet, so I haven't gone mad.

### CODE REVIEW

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift
Comment (gclssvglx) [inline Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:16]: Interesting, I was going to ask if there is a way to get some tolerance on this. But, I totally get that we want pixel-perfect 👍

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Specs/ViewControllers/TestViewControllerSnapshotTests.swift
Comment (gclssvglx) [inline Tests/govuk_ios/govuk_ios_snapshot_tests/Specs/ViewControllers/TestViewControllerSnapshotTests.swift:10]: Nice that you don't have to specify RGB or CMYK values here 🥇

File: GovUK.xcodeproj/project.pbxproj
Comment (gclssvglx) [inline GovUK.xcodeproj/project.pbxproj:1]: These files always make me 🤮  I guess I'll get used to them 😉

File: Tests/govuk_ios/govuk_ios_snapshot_tests/ReferenceImages_64/govuk_ios_snapshot_tests.LaunchViewControllerTests/test_loadInNavigationController_rendersCorrectly@3x.png
Comment (gclssvglx) [inline Tests/govuk_ios/govuk_ios_snapshot_tests/ReferenceImages_64/govuk_ios_snapshot_tests.LaunchViewControllerTests/test_loadInNavigationController_rendersCorrectly@3x.png:1]: Be interesting to see how adding these images and GitHub LFS impacts `pull`s

File: Tests/govuk_ios/shared/SupportingFiles/TestAppDelegate.swift
Comment (gclssvglx) [inline Tests/govuk_ios/shared/SupportingFiles/TestAppDelegate.swift:7]: Interesting, if set to `true` would it record video?

**Review by gclssvglx — APPROVED** [review]
Can't speak for the iOS-specific stuff sorry, but the overall plan looks good. I've a few questions - no blockers.

FYI - Looks like 'native' screenshot testing is [being introduced in Android Studio soon](https://developer.android.com/studio/preview/compose-screenshot-testing).

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift
Comment (robinphillips) [inline Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:3]: Is there an approval process required to use this? I think if we are using it we will need to include the MIT licence.

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift
Comment (batesyuk) [reply Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:16]: We probably won't want pixel perfect forever, but until it breaks, might as well start there

File: GovUK.xcodeproj/project.pbxproj
Comment (batesyuk) [reply GovUK.xcodeproj/project.pbxproj:1]: You won't

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift
Comment (batesyuk) [reply Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:3]: We are going to need to include all 3rd party libs licences and acknowledgements, yeah. Not sure we will need much approval to use it in general given its not production code. Worth checking before merging though.

**Review by 1pretz1 — APPROVED** [review]
Looks good to me 👍

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift
Comment (raphaelolu) [inline Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:11]: A comment appears to be left in by accident

**Review by raphaelolu — CHANGES_REQUESTED** [review]

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift
Comment (batesyuk) [reply Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:11]: This is deliberate. Its to uncomment when/ if we want to retake all snapshot tests.

**Review by raphaelolu — APPROVED** [review]

**Pull request merged**
